From 478324efc227027184ca6b4716803d03bdeab033 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 28 Mar 2018 04:37:20 -0700
Subject: [PATCH 32/41] Add LOG_WARNING_MESSAGE to RInit.cpp initialize

This always causes me problems and it is usually something dumb like files not
being in the right place during debugging sessions (where I'm probably trying to
debug something else, but this should reduce the amount of time I waste).

The messages will appear in %AppLocalData%\RStudio-Desktop\log\rsession-builder.log
---
 src/cpp/r/session/RInit.cpp | 22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/src/cpp/r/session/RInit.cpp b/src/cpp/r/session/RInit.cpp
index 7a945388ce..d12eb24fde 100644
--- a/src/cpp/r/session/RInit.cpp
+++ b/src/cpp/r/session/RInit.cpp
@@ -202,14 +202,19 @@ Error initialize()
    // ensure that the utils package is loaded (it might not be loaded
    // if R is attempting to recover from a library loading error which
    // occurs during .Rprofile)
+   LOG_WARNING_MESSAGE("initialize 1");
    Error libError = r::exec::RFunction("library", "utils").call();
+   LOG_WARNING_MESSAGE("initialize 2");
    if (libError)
       LOG_ERROR(libError);
+   LOG_WARNING_MESSAGE("initialize 3");
 
    // check whether this is R 3.3 or greater
    Error r33Error = r::exec::evaluateString("getRversion() >= '3.3.0'", &s_isR3_3);
+   LOG_WARNING_MESSAGE("initialize 4");
    if (r33Error)
       LOG_ERROR(r33Error);
+   LOG_WARNING_MESSAGE("initialize 5");
 
    if (s_isR3_3)
    {
@@ -223,6 +228,7 @@ Error initialize()
          LOG_ERROR(r3Error);
    }
 
+   LOG_WARNING_MESSAGE("initialize 6");
    // initialize console history capacity
    r::session::consoleHistory().setCapacityFromRHistsize();
 
@@ -234,9 +240,11 @@ Error initialize()
 
    // install RStudio API
    FilePath apiFilePath = utils::rSourcePath().completePath("Api.R");
+   LOG_WARNING_MESSAGE("initialize 7");
    error = r::sourceManager().sourceTools(apiFilePath);
    if (error)
       return error;
+   LOG_WARNING_MESSAGE("initialize 8");
 
    // initialize graphics device -- use a stable directory for server mode
    // and temp directory for desktop mode (so that we can support multiple
@@ -254,6 +262,7 @@ Error initialize()
       graphicsPath = r::session::utils::tempDir().completePath(
          "rs-graphics-" + core::system::generateUuid());
    }
+   LOG_WARNING_MESSAGE("initialize 9");
 
    error = graphics::device::initialize(graphicsPath,
                                         rCallbacks().locator);
@@ -263,13 +272,15 @@ Error initialize()
    // restore client state
    session::clientState().restore(utils::clientStatePath(),
                                   utils::projectClientStatePath());
-      
+
+   LOG_WARNING_MESSAGE("initialize 10");
    // restore suspended session if we have one
    bool wasResumed = false;
    
    // first check for a pending restart
    if (restartContext().hasSessionState())
    {
+      LOG_WARNING_MESSAGE("initialize 11");
       // restore session
       std::string errorMessages;
       restoreSession(restartContext().sessionStatePath(), &errorMessages);
@@ -284,7 +295,8 @@ Error initialize()
    else if (suspendedSessionPath().exists())
    {  
       // restore session
-      std::string errorMessages;
+      LOG_WARNING_MESSAGE("initialize 12");
+      std::string errorMessages ;
       restoreSession(suspendedSessionPath(), &errorMessages);
       
       // show any error messages
@@ -296,7 +308,8 @@ Error initialize()
    }  
    // new session
    else
-   {  
+   {
+      LOG_WARNING_MESSAGE("initialize 13");
       // restore console history
       FilePath historyPath = rHistoryFilePath();
       error = consoleHistory().loadFromFile(historyPath, false);
@@ -308,6 +321,7 @@ Error initialize()
    }
    
    // initialize client
+   LOG_WARNING_MESSAGE("initialize 14");
    RInitInfo rInitInfo(wasResumed);
    error = rCallbacks().init(rInitInfo);
    if (error)
@@ -331,6 +345,7 @@ Error initialize()
       if (error)
          return error;
    }
+   LOG_WARNING_MESSAGE("initialize 15");
 
    // initialize profile resources
    error = r::exec::RFunction(".rs.profileResources").call();
@@ -348,6 +363,7 @@ Error initialize()
    if (error)
       return error;
 
+   LOG_WARNING_MESSAGE("initialize 16");
    // server specific R options options
    if (utils::isServerMode())
    {
-- 
2.29.2.windows.3

